docs/matrix.md

```diff
@@ -172,14 +179,256 @@ The `Matrix.Include` will take `Parameters` of type `"string"` only.
   ...
 ```
 
+#### Include specific combinations in the Matrix
+
+In the example below, the `Pipeline` defines a `Matrix` that will execute golang-test in three different linux flavors.
+
+While running on amd64 and ppc64le, golang-test should execute `go test github.com/tektoncd/pipeline/pkg/...`.
+While running on linux/s390x, golang-test should execute `go test github.com/tektoncd/pipeline/pkg/... -cover -v`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline-to-build-and-test-go-project
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - ...
+    - name: golang-test
+      taskRef:
+        name: golang-test
+      params:
+        - name: package
+          value: "github.com/tektoncd/pipeline"
+        - name: packages
+          value: "./pkg/..."
+      matrix:
+        params:
+          - name: GOARCH
+            value:
+              - "linux/amd64"
+              - "linux/ppc64le"
+              - "linux/s390x"
+        include:
+          - name: s390x-no-race
+            params:
+              - name: GOARCH
+                value: "linux/s390x"
+              - name: flags
+                value: "-cover -v"
+    - ...
+```
+
+The above `Matrix` specification will create three `TaskRuns`, one for each architecture in GOARCH ("linux/amd64", "linux/ppc64le", and "linux/s390x") and with default flags. Now, when a `TaskRun` is created for the GOARCH value of linux/s390x, an additional `Parameter` called flags with the value of `-cover -v` will be included in the taskRun.
+
+```json!
+{ "GOARCH": "linux/amd64", "version": "go1.17" }
+
+{ "GOARCH": "linux/amd64", "version": "go1.18.1" }
+
+{ "GOARCH": "linux/ppc64le", "version": "go1.17" }
+
+{ "GOARCH": "linux/ppc64le", "version": "go1.18.1" }
+
+{ "GOARCH": "linux/s390x", "version": "go1.17", "flags": "-cover -v" }
+
+{ "GOARCH": "linux/s390x", "version": "go1.18.1", "flags": "-cover -v" }
+
+```
+
+The `Matrix.Include` section can list params which does not exist in the matrix.params section. In this example specification, flags was not listed in `Matrix.Params`. At the same time, `Matrix.Include` section can list params which does exist in the `Matrix.Params` section.
```

Comment:
This line is a bit confusing/surprising to see in the middle of an example. What I'd do instead is in the section where `include` is first introduced, say something like:
"Each element of matrix.include is a combination of string parameters to use when creating a TaskRun. Paramers in `matrix.include` don't need to be specified in `matrix.params`; see [example] for an example. If parameters in `matrix.include` are also listed in matrix.params, the combinations used for fanned out TaskRuns depend on whether the values of these parameters are the same in matrix.include and matrix.params. see [example] for more details"